javalib/src/main/scala/java/nio/charset/Charset.scala

```diff
@@ -8,6 +8,9 @@ abstract class Charset protected (canonicalName: String,
     aliases: Array[String]) extends AnyRef with Comparable[Charset] {
   final def name(): String = canonicalName
 
+  final def aliases(): java.util.Set[String] =
+    new java.util.HashSet(java.util.Arrays.asList(aliases))
```

Comment:
The Javadoc says that `aliases()` must return an *immutable* set of the aliases. Therefore, you should wrap the produced `HashSet` with [`java.util.Collections.unmodifiableSet`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#unmodifiableSet-java.util.Set-). This also means that you can always reuse the same unmodifiable set, which you can store in a `private lazy val aliasesSet`. Make sure to make it lazy so that all this stuff can be dce'ed away if `aliases()` is not called.